services/cloudrun-service/pkg/cloudrun/cloudrun.go

```diff
@@ -0,0 +1,186 @@
+	}
+	if p.locationMissing {
+		return fmt.Sprintf("location is missing for service: %s", p.name)
+	}
+	return "Configuration error"
+}
+
+func Init(ctx context.Context) error {
+	var err error
+	runService, err = run.NewService(ctx)
+	return err
+}
+
+func Deploy(ctx context.Context, svc *run.Service) error {
+	serviceName := svc.Metadata.Name
+	// Get the full path of the project. Example: projects/<project-id>/locations/<region>
+	parent, err := getParent(svc)
+	servicePath := fmt.Sprintf("%s/services/%s", parent, serviceName)
+	if err != nil {
+		return err
+	}
+	req := runService.Projects.Locations.Services.List(parent)
+	resp, err := req.Do()
+	if err != nil {
+		return err
+	}
+	var serviceCallResp *run.Service
+	// If the service is already deployed before, then we need to call ReplaceService. Otherwise, we call Create.
+	if isPresent(resp.Items, serviceName) {
+		serviceCall := runService.Projects.Locations.Services.ReplaceService(servicePath, svc)
+		serviceCallResp, err = serviceCall.Do()
+		if err != nil {
+			return err
+		}
+	} else {
+		serviceCall := runService.Projects.Locations.Services.Create(parent, svc)
+		serviceCallResp, err = serviceCall.Do()
+		if err != nil {
+			return err
+		}
+	}
+	if err := waitForOperation(parent, serviceCallResp, 60); err != nil {
+		return err
+	}
+	getServiceCall := runService.Projects.Locations.Services.Get(servicePath)
+	serviceResp, err := getServiceCall.Do()
+	if err != nil {
+		return err
+	}
+	condition, err := GetServiceReadyCondition(serviceResp)
+	if err != nil {
+		return err
+	}
+	if condition.Status != "True" {
+		return fmt.Errorf("service not ready: %s", condition)
+	}
+	return nil
+}
+
+func GetServiceReadyCondition(serviceCallResponse *run.Service) (ServiceReadyCondition, error) {
+	//exhaustruct:ignore
+	serviceReadyCondition := ServiceReadyCondition{
+		Status:   "",
+		Name:     serviceCallResponse.Metadata.Name,
+		Revision: serviceCallResponse.Status.ObservedGeneration,
+	}
+	conditions := serviceCallResponse.Status.Conditions
+	for _, condition := range conditions {
+		if condition.Type == serviceReady {
+			serviceReadyCondition.Status = condition.Status
+			serviceReadyCondition.Reason = condition.Reason
+			serviceReadyCondition.Message = condition.Message
+		}
+	}
+	if serviceReadyCondition.Status == "" {
+		return serviceReadyCondition, fmt.Errorf("failed to get Ready status for service %s", serviceReadyCondition.Name)
+	}
+	return serviceReadyCondition, nil
+}
+
+func getOperationId(parent string, serviceCallResp *run.Service) (string, error) {
+	operationId, exists := serviceCallResp.Metadata.Annotations[operationIdAnnotation]
+	if !exists {
+		return "", fmt.Errorf("failed to get operation-id for service %s", serviceCallResp.Metadata.Name)
+	}
+	return fmt.Sprintf("%s/operations/%s", parent, operationId), nil
+}
+
+func waitForOperation(parent string, serviceCallResp *run.Service, timeoutSeconds uint16) error {
```

Comment:
please use duration instead of an int
```suggestion
func waitForOperation(parent string, serviceCallResp *run.Service, timeout time.Duration) error {
```
https://pkg.go.dev/time#pkg-types